lib/xgu/xgu.h

```diff
@@ -0,0 +1,937 @@
+    XGU_FACTOR_DST_COLOR = NV097_SET_BLEND_FUNC_SFACTOR_V_DST_COLOR,
+    XGU_FACTOR_ONE_MINUS_DST_COLOR = NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_DST_COLOR,
+    XGU_FACTOR_SRC_ALPHA_SATURATE = NV097_SET_BLEND_FUNC_SFACTOR_V_SRC_ALPHA_SATURATE,
+    XGU_FACTOR_CONSTANT_COLOR = NV097_SET_BLEND_FUNC_SFACTOR_V_CONSTANT_COLOR,
+    XGU_FACTOR_ONE_MINUS_CONSTANT_COLOR = NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_CONSTANT_COLOR,
+    XGU_FACTOR_CONSTANT_ALPHA = NV097_SET_BLEND_FUNC_SFACTOR_V_CONSTANT_ALPHA,
+    XGU_FACTOR_ONE_MINUS_CONSTANT_ALPHA = NV097_SET_BLEND_FUNC_SFACTOR_V_ONE_MINUS_CONSTANT_ALPHA
+} XguBlendFactor;
+
+typedef enum {
+    XGU_BLUE = NV097_SET_COLOR_MASK_BLUE_WRITE_ENABLE,
+    XGU_GREEN = NV097_SET_COLOR_MASK_GREEN_WRITE_ENABLE,
+    XGU_RED = NV097_SET_COLOR_MASK_RED_WRITE_ENABLE,
+    XGU_ALPHA = NV097_SET_COLOR_MASK_ALPHA_WRITE_ENABLE
+} XguColorMask;
+
+typedef enum {
+    XGU_POINT = NV097_SET_FRONT_POLYGON_MODE_V_POINT,
+    XGU_LINE = NV097_SET_FRONT_POLYGON_MODE_V_LINE,
+    XGU_FILL = NV097_SET_FRONT_POLYGON_MODE_V_FILL
+} XguPolygonMode;
+
+typedef enum {
+    XGU_CULL_FRONT = NV097_SET_CULL_FACE_V_FRONT,
+    XGU_CULL_BACK = NV097_SET_CULL_FACE_V_BACK,
+    XGU_CULL_FRONT_AND_BACK = NV097_SET_CULL_FACE_V_FRONT_AND_BACK
+} XguCullFace;
+
+typedef enum {
+    XGU_TEXGEN_DISABLE = NV097_SET_TEXGEN_S_DISABLE,
+    XGU_TEXGEN_EYE_LINEAR = NV097_SET_TEXGEN_S_EYE_LINEAR,
+    XGU_TEXGEN_OBJECT_LINEAR = NV097_SET_TEXGEN_S_OBJECT_LINEAR,
+    XGU_TEXGEN_SPHERE_MAP = NV097_SET_TEXGEN_S_SPHERE_MAP,
+    XGU_TEXGEN_REFLECTION_MAP = NV097_SET_TEXGEN_S_REFLECTION_MAP,
+    XGU_TEXGEN_NORMAL_MAP = NV097_SET_TEXGEN_S_NORMAL_MAP
+} XguTexgen;
+
+typedef enum {
+    XGU_SOURCE_TEXTURE = NV097_SET_TEXTURE_FORMAT_BORDER_SOURCE_TEXTURE,
+    XGU_SOURCE_COLOR = NV097_SET_TEXTURE_FORMAT_BORDER_SOURCE_COLOR
+} XguBorderSrc;
+
+typedef enum {
+    XGU_TEXTURE_FORMAT_COLOR_SZ_Y8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_Y8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_AY8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_AY8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_A1R5G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A1R5G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_X1R5G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_X1R5G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_A4R4G4B4 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A4R4G4B4,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_R5G6B5 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_R5G6B5,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_X8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_X8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_SZ_I8_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_SZ_I8_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_L_DXT1_A1R5G5B5 = NV097_SET_TEXTURE_FORMAT_COLOR_L_DXT1_A1R5G5B5,
+    XGU_TEXTURE_FORMAT_COLOR_L_DXT23_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_L_DXT23_A8R8G8B8,
+    XGU_TEXTURE_FORMAT_COLOR_L_DXT45_A8R8G8B8 = NV097_SET_TEXTURE_FORMAT_COLOR_L_DXT45_A8R8G8B8,
```

Comment:
these DXT formats are compressed formats, we should probably mention this and remove the `L_` instead to keep the name predictable.
We probably also want to rename them to DXT1, DXT3 and DXT5, possibly using aliases for DXT2 and DXT4 (which are less common formats of DXT3 and DXT5 - I'd have to look up the differences - we can possibly ignore them)